File: js_solutions/0504_base7.js
/**
 * @link https://leetcode.com/problems/base-7/submissions/
 * @description Return a string of number's base 7 representation.
 * @param {Number} number 
 * @param {Number} base 
 * @returns {String}
 */
 const convertToBase7 = (number, base = 7) => {
  // Case 1: Return 0, if number equals 0
  if(number === 0) {
    return `${0}`;
  }
  let absValue = Math.abs(number);
  let convertedNumber = '';

  // Case 2: Use Gorner's Algorithm to convert number 
  while(absValue > 0) {
    const lastNumber = absValue % base; // Последнее число
    absValue = Math.trunc(absValue / base);
    convertedNumber = `${lastNumber}${convertedNumber}`;
  }
  return number < 0 ? -`${convertedNumber}` : convertedNumber;
};

File: python_solutions/0504_base7.py
def convertToBase7(self, num: int) -> str:
  # Case 1: Return 0, if number equals 0
  if (num == 0):
    return str(0)
        
  absValue = abs(num)
  convertedNumber = ''
        
  # Case 2: Use Gorner's Algorithm to convert number
  while (absValue > 0):
    # Take last digit of given number
    lastDigit = absValue % 7 
    absValue = absValue // 7
    convertedNumber = str(lastDigit) + convertedNumber
        
  return '-' + str(convertedNumber) if num < 0 else convertedNumber